rust/src/ssh/detect.rs

```diff
@@ -81,7 +80,6 @@ pub extern "C" fn rs_ssh_tx_get_software(
                 return 1;
             }
         }
-        _ => {}
```

Comment:
@catenacyber this change makes `ssh-banner` test fail. direction was 6 in that case and it was bypassed earlier but now with the Direction enum, we strictly check for 0x04 (Server) and 0x08 (Client) in flags and convert them if they are something other than these two values.
So, in this case, 6 was converted to 0x04 (ToServer) and that makes the test fail.

Please note that this conversion part is done because flags can carry a lot other info other than direction. ref: https://github.com/OISF/suricata/pull/6294/commits/dbfbeb9aac88be32373e950bf4775e1a26867919#diff-56528694f6f3213d6fb88d872f77291412dceec263b57166519843b13eca9a4dR56

Reply:
@inashivb 
This looks like a bug in the current SSH code
Is `ssh-banner` a Suricata-verify test ? How does it fail ?
Why did you remove these SSH changes in the next version of the PR ? cf #6302

Reply:
And the SSH bug was reported as https://redmine.openinfosecfoundation.org/issues/4563

Reply:
Thank you, @catenacyber ! I had removed it from the next version of PR to get what works merged first. But, if ssh fix gets merged, I'll add that to my next revision.

Reply:
Oh, I just checked. So, the test needs to be fixed and this is OK?

Reply:
Yes, the Suricata-verify test needs to be fixed (and the code as well as you did)